pkg/inject/inject.go

```diff
@@ -666,7 +663,19 @@ func (conf *ResourceConfig) injectPodSpec(values *podPatch) {
 func (conf *ResourceConfig) injectProxyInit(values *podPatch) {
 
 	// Fill common fields from Proxy into ProxyInit
-	values.ProxyInit.Capabilities = values.Proxy.Capabilities
+	values.ProxyInit.Capabilities = &l5dcharts.Capabilities{}
+	if values.Proxy.Capabilities != nil {
```

Comment:
@aryan9600 I'm realizing we only need to initialize `values.ProxyInit.Capabilities` if `values.Proxy.Capabilities != nil`:

```
  if values.Proxy.Capabilities != nil {
      values.ProxyInit.Capabilities = &l5dcharts.Capabilities{}
```